test/compiler/inference.jl

```diff
@@ -3403,3 +3403,10 @@ end
         @test @inferred(f40177(T)) == fieldtype(T, 1)
     end
 end
+
+# issue #41908
+struct Diagonalish{T,V<:AbstractVector{T}} <: AbstractMatrix{T}
+end
+f41908(x::Diagonalish{Int, T}) where {Int<:T<:Int} = 1
```

Comment:
The `Diagonal` thing was just the first that came to mind in #41908, we could just as well use `Complex{String}` here.
```suggestion
f41908(x::Complex{T}) where {String<:T<:String} = 1
```
Not that it matters much, but when in doubt, I tend to prefer simpler test cases.